### ISSUE TITLE
Debugger stops twice when hitting continue after stopping at a breakpoint

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/debugpy#964]
Before creating a new issue, please check the [FAQ](https://github.com/microsoft/debugpy/wiki/FAQ) to see if your question is answered there.

## Environment data

- debugpy version: 1.6.0 (run `import debugpy; print(debugpy.__version__)` if uncertain)
- OS and version: Mac
- Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.13 Virtual Env
- Using VS Code or Visual Studio: VS Code

## Actual behavior

Debugger stops twice when hitting continue after stopping at a breakpoint

## Expected  behavior

Should continue instead of stopping at the same place again.

## Steps to reproduce:
1. Create a Jupyter notebook
2. Add a cell with the code
```python
print(1)
print(2)
print(3)
print(4)
```
3. Add a breakpoint to the second & third lines
4. Ensure focus is set to this cell.
5. Start debugging using the command `Debug Cell`
Notice how the debugger now stops at the second line as expected, now hit the `Continue` button, & note how the debugger stops again at the exact same line.
Continue again & it will continue as expected to line 3, but continuing will stop in line 3 yet again.

Basically the debugger stops twice for every breakpoint.

**Comment by fabioz** [comment]
Hello @DonJayamanne,

I'm closing this issue because the issues related to Jupyter are being tracked at:
https://github.com/microsoft/debugpy/issues/869

In particular, the most clarifying comment about it is probably (the real issue is that Jupyter doesn't really execute things as a single frame, rather it does an exec line by line and the debugger needs to understand this better to handle it appropriately):
https://github.com/microsoft/debugpy/issues/869#issuecomment-1132141003
